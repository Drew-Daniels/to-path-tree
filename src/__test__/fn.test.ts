import { pathToTree, walkPathTree } from '..'
import input from './fixtures/input.json'

describe('test', () => {
  it('paths to tree', () => {
    expect(pathToTree(input)).toMatchSnapshot()
  })
  it('walker', () => {
    const tree = pathToTree(input)
    walkPathTree(tree, (node) => {
      for (const item of node.items) {
        if (item.path === 'src/a/b/d/index.ts')
          expect(node.parent?.items).toMatchSnapshot()
      }
    })
  })
})

it('should add directories', () => {
  const tree = pathToTree([
    '/a/',
    '/a/b/',
  ])
  expect(tree).toMatchSnapshot()
})
