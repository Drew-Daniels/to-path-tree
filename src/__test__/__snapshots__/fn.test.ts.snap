// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should add directories 1`] = `
{
  "items": [],
  "name": "root",
  "subDirectory": {
    "a": {
      "items": [],
      "name": "a",
      "parent": [Circular],
      "subDirectory": {
        "b": {
          "items": [],
          "name": "b",
          "parent": [Circular],
          "subDirectory": {},
        },
      },
    },
  },
}
`;

exports[`test > paths to tree 1`] = `
{
  "items": [],
  "name": "root",
  "subDirectory": {
    "a": {
      "items": [
        {
          "data": undefined,
          "ext": "ts",
          "filename": "index",
          "isEntry": true,
          "parent": [Circular],
          "path": "a/index.ts",
        },
        {
          "data": undefined,
          "ext": "ts",
          "filename": "b",
          "isEntry": false,
          "parent": [Circular],
          "path": "a/b.ts",
        },
      ],
      "name": "a",
      "parent": [Circular],
      "subDirectory": {
        "b": {
          "items": [
            {
              "data": undefined,
              "ext": "js",
              "filename": "index",
              "isEntry": true,
              "parent": [Circular],
              "path": "a/b/index.js",
            },
          ],
          "name": "b",
          "parent": [Circular],
          "subDirectory": null,
        },
      },
    },
    "src": {
      "items": [
        {
          "data": undefined,
          "ext": "ts",
          "filename": "index",
          "isEntry": true,
          "parent": [Circular],
          "path": "src/index.ts",
        },
        {
          "data": undefined,
          "ext": "ts",
          "filename": "foo",
          "isEntry": false,
          "parent": [Circular],
          "path": "src/foo.ts",
        },
      ],
      "name": "src",
      "parent": [Circular],
      "subDirectory": {
        "a": {
          "items": [
            {
              "data": undefined,
              "ext": "ts",
              "filename": "index",
              "isEntry": true,
              "parent": [Circular],
              "path": "src/a/index.ts",
            },
          ],
          "name": "a",
          "parent": [Circular],
          "subDirectory": {
            "b": {
              "items": [
                {
                  "data": undefined,
                  "ext": "ts",
                  "filename": "index",
                  "isEntry": true,
                  "parent": [Circular],
                  "path": "src/a/b/index.ts",
                },
              ],
              "name": "b",
              "parent": [Circular],
              "subDirectory": {
                "d": {
                  "items": [
                    {
                      "data": undefined,
                      "ext": "ts",
                      "filename": "index",
                      "isEntry": true,
                      "parent": [Circular],
                      "path": "src/a/b/d/index.ts",
                    },
                  ],
                  "name": "d",
                  "parent": [Circular],
                  "subDirectory": null,
                },
              },
            },
          },
        },
        "b": {
          "items": [
            {
              "data": undefined,
              "ext": "js",
              "filename": "index",
              "isEntry": true,
              "parent": [Circular],
              "path": "src/b/index.js",
            },
          ],
          "name": "b",
          "parent": [Circular],
          "subDirectory": {
            "c": {
              "items": [
                {
                  "data": undefined,
                  "ext": "ts",
                  "filename": "index",
                  "isEntry": true,
                  "parent": [Circular],
                  "path": "src/b/c/index.ts",
                },
              ],
              "name": "c",
              "parent": [Circular],
              "subDirectory": null,
            },
          },
        },
      },
    },
  },
}
`;

exports[`test > walker 1`] = `
[
  {
    "data": undefined,
    "ext": "ts",
    "filename": "index",
    "isEntry": true,
    "parent": {
      "items": [Circular],
      "name": "b",
      "parent": {
        "items": [
          {
            "data": undefined,
            "ext": "ts",
            "filename": "index",
            "isEntry": true,
            "parent": [Circular],
            "path": "src/a/index.ts",
          },
        ],
        "name": "a",
        "parent": {
          "items": [
            {
              "data": undefined,
              "ext": "ts",
              "filename": "index",
              "isEntry": true,
              "parent": [Circular],
              "path": "src/index.ts",
            },
            {
              "data": undefined,
              "ext": "ts",
              "filename": "foo",
              "isEntry": false,
              "parent": [Circular],
              "path": "src/foo.ts",
            },
          ],
          "name": "src",
          "parent": {
            "items": [],
            "name": "root",
            "subDirectory": {
              "a": {
                "items": [
                  {
                    "data": undefined,
                    "ext": "ts",
                    "filename": "index",
                    "isEntry": true,
                    "parent": [Circular],
                    "path": "a/index.ts",
                  },
                  {
                    "data": undefined,
                    "ext": "ts",
                    "filename": "b",
                    "isEntry": false,
                    "parent": [Circular],
                    "path": "a/b.ts",
                  },
                ],
                "name": "a",
                "parent": [Circular],
                "subDirectory": {
                  "b": {
                    "items": [
                      {
                        "data": undefined,
                        "ext": "js",
                        "filename": "index",
                        "isEntry": true,
                        "parent": [Circular],
                        "path": "a/b/index.js",
                      },
                    ],
                    "name": "b",
                    "parent": [Circular],
                    "subDirectory": null,
                  },
                },
              },
              "src": [Circular],
            },
          },
          "subDirectory": {
            "a": [Circular],
            "b": {
              "items": [
                {
                  "data": undefined,
                  "ext": "js",
                  "filename": "index",
                  "isEntry": true,
                  "parent": [Circular],
                  "path": "src/b/index.js",
                },
              ],
              "name": "b",
              "parent": [Circular],
              "subDirectory": {
                "c": {
                  "items": [
                    {
                      "data": undefined,
                      "ext": "ts",
                      "filename": "index",
                      "isEntry": true,
                      "parent": [Circular],
                      "path": "src/b/c/index.ts",
                    },
                  ],
                  "name": "c",
                  "parent": [Circular],
                  "subDirectory": null,
                },
              },
            },
          },
        },
        "subDirectory": {
          "b": [Circular],
        },
      },
      "subDirectory": {
        "d": {
          "items": [
            {
              "data": undefined,
              "ext": "ts",
              "filename": "index",
              "isEntry": true,
              "parent": [Circular],
              "path": "src/a/b/d/index.ts",
            },
          ],
          "name": "d",
          "parent": [Circular],
          "subDirectory": null,
        },
      },
    },
    "path": "src/a/b/index.ts",
  },
]
`;
